cpp/src/io/parquet/writer_impl.cu

```diff
@@ -384,15 +384,34 @@ writer::impl::impl(std::unique_ptr<data_sink> sink, writer_options const &option
   stats_granularity_(options.stats_granularity),
   out_sink_(std::move(sink)){}
 
-void writer::impl::write(table_view const &table, const table_metadata *metadata, cudaStream_t stream) {
+std::unique_ptr<std::vector<uint8_t>> writer::impl::write(
+                         table_view const &table, const table_metadata *metadata,
+                         bool return_filemetadata, const std::string metadata_out_file_path,
+                         cudaStream_t stream) {    
   pq_chunked_state state;
   state.user_metadata = metadata;
   state.stream = stream;
   state.single_write_mode = true;
 
   write_chunked_begin(state);
   write_chunked(table, state);
-  write_chunked_end(state);    
+  write_chunked_end(state);
+
+  // Optionally output raw file metadata with the specified column chunk file path
+  if (return_filemetadata) {
+    CompactProtocolWriter cpw(&buffer_);
+    buffer_.resize(0);
+    for (auto& rowgroup : state.md.row_groups) {
+      for (auto& col : rowgroup.columns) {
+        col.file_path = metadata_out_file_path;
+      }
+    }
+    cpw.write(&state.md);
+    return std::make_unique<std::vector<uint8_t>>(std::move(buffer_));
+  }
+  else {
+    return {nullptr};
+  }
```

Comment:
This seems like it should be refactored into write_chunked_end() and the relevant parameters exposed to write_parquet_chunked_args.   Seems like there's no technical reason this can't be exposed to both write paths.

Reply:
I was thinking that as well at first, but then realized that we're not actually returning the same metadata that what is written in the file (due to column chunk's file_path), so this is something that happens after write_chunked_end (which is still called in the non-chunked code path). Also, while it would be possible to do so, it actually does not make much sense to use this functionality along with the chunked writing since the goal here is to write to many different files, whereas write_chunked is for many writes to a single file (in the latter case, the metadata is the same as the one written to the file) ~ this is really dask-specific.

Reply:
That's true, but at the same time, write is effectively just a simple wrapper around write_chunked. That is to say :  write_chunked is actually where all the implementation is.  So even if in the write_chunked case we'd likely never be doing this, we -might- someday want to do it.  Having that interface available doesn't affect the immediate feature need, it just keeps the door open for other stuff.

Reply:
I see what you mean, though that would mean passing in dummy return_filemetadata=false and an empty metadata_out_file_path parameters to write_chunked_end ?

Reply:
Ok moved it to write_chunked_end()